mkosi/config.py

```diff
@@ -1964,6 +1964,11 @@ def load_config(args: argparse.Namespace) -> MkosiConfig:
             output = prefix
         args.output = Path(output)
 
+    if "_" in str(args.output) and args.output_format == OutputFormat.subvolume:
+        old = str(args.output)
+        args.output = args.output.with_name(args.output.name.replace("_", "-"))
+        logging.warning(f"subvolume (systemd-machined) names cannot contain underscore(_); converting to hypen (-). {old} -> {args.output} ")
```

Comment:
I was confused by your first comment since I see:
```
‣ subvolume (systemd-machined) names cannot contain underscore(_); converting to hypen (-). debian_bullseye_amd64_0.2 -> debian-bullseye-amd64-0.2 
```
but then realised you are referring to `output` containing a path component, whereas in my scenario it is created from `ImageId` and `ImageVersion`. So, would need to do this only on the last component in the path.

Typo for "hyphen" - my brain and fingers have different ideas on spelling!

I'd rather it doesn't fail since the same configuration and naming may be used with different `output_format` s and a fail would involve having to change the config just for the `subvolume` build. E.g. in scenarios where builds use each format in turn, in a loop.

Also note that currently when `image_version` is added to `prefix` it inserts an underscore and I was trying to avoid changing that as well (originally I was replacing that underscore with "-v").

Reply:
OK. Then change the warning to a info or debug message.